pkg/client/client.go

```diff
@@ -1053,10 +1054,11 @@ func (c *Client) CreateOrUpdateRoleBinding(rb *rbacv1.RoleBinding) error {
 		return errors.Wrap(err, "retrieving RoleBinding object failed")
 	}
 
-	if reflect.DeepEqual(rb.RoleRef, r.RoleRef) &&
-		reflect.DeepEqual(rb.Subjects, r.Subjects) &&
-		reflect.DeepEqual(rb.Labels, r.Labels) &&
-		reflect.DeepEqual(rb.Annotations, r.Annotations) {
+	mergo.Merge(&rb.Annotations, existing.Annotations)
```

Comment:
this won't remove old annotations. Is that ok?

At least this must be godoc'ed on this func.

Reply:
Also this mutates rb.Annotation.

/hold

Reply:
Yes, this won't remove old annotations. The goal was to merge old/existing ones with what is defined by the operator.

Since `rb` and `crb` is what we hardcode in operator, mutating `rb.Annotation` is how we can update whole object with added Annotations.

I have added a comment to reflect that.

Reply:
it is very bad style to have side-effects in such a function, especially when named `CreateOrUpdateRoleBinding`. None of the upstream client funcs does that for a reason. This func should not divert. It is even dangerious if it does because consumers will not expect it and potentially invalidate objects from a cache.

Reply:
I agree here, for the caller this is unexpected. @paulfantom let's make sure we address this in the follow-up. An easy way forward would be to create a deep copy of `rb`.